src/go/pt-k8s-debug-collector/dumper/dumper.go

```diff
@@ -53,56 +63,72 @@ func New(location, namespace, resource string, kubeconfig string, forwardport st
 		"persistentvolumeclaims",
 		"persistentvolumes",
 	}
-	filePaths := make([]string, 0)
-	if len(resource) > 0 {
-		if resourceType(resource) == "pxc" {
-			resources = append(resources,
-				"perconaxtradbclusterbackups",
-				"perconaxtradbclusterrestores",
-				"perconaxtradbclusters")
-			filePaths = append(filePaths,
-				"var/lib/mysql/mysqld-error.log",
-				"var/lib/mysql/innobackup.backup.log",
-				"var/lib/mysql/innobackup.move.log",
-				"var/lib/mysql/innobackup.prepare.log",
-				"var/lib/mysql/grastate.dat",
-				"var/lib/mysql/gvwstate.dat",
-				"var/lib/mysql/mysqld.post.processing.log",
-				"var/lib/mysql/auto.cnf",
-			)
-		} else if resourceType(resource) == "psmdb" {
-			resources = append(resources,
-				"perconaservermongodbbackups",
-				"perconaservermongodbrestores",
-				"perconaservermongodbs",
-			)
-		} else if resourceType(resource) == "pg" {
-			resources = append(resources,
-				"perconapgclusters",
-				"pgclusters",
-				"pgpolicies",
-				"pgreplicas",
-				"pgtasks",
-			)
-		} else if resourceType(resource) == "ps" {
-			resources = append(resources,
-				"perconaservermysqlbackups",
-				"perconaservermysqlrestores",
-				"perconaservermysqls",
-			)
+
+	switch resource {
+	case "auto":
+		result, err := d.runCmd("api-resources", "-o", "name")
+		if err != nil {
+			log.Panicf("Cannot get API resources and option --resource=auto specified:\n%s", err)
+		}
+		matches := resourcesRe.FindAllStringSubmatch(string(result), -1)
+		if len(matches) == 0 {
+			resource = "none"
+			break
 		}
+		for _, match := range matches {
+			resources = append(resources, match[1])
+			resource = match[2]
```

Comment:
Not sure if I'm getting it right, but:
does it mean "auto" will try to match a single resource ?

If it matches both pxc+pg, resource will (maybe) trigger on pxc, then pg, so resource will be overwritten with "pg"

Later on, it means it won't be adding the proper resources down the line ?

What I'm describing would only be useful if customer create different db in the same namespace, but here's my counter proposal:
creating a map such as
```
type resourceData struct {
  k8sResourcesToCheck []string
  fileToCheck []string
}
var resourcesMap map[string]resourceData
```
Later on, use that as
```
for _, match := range matches {
  resources = append(resources, resourcesMap[match[1]].k8sResourcesToCheck...)
  filePaths = append(resources, resourcesMap[match[1]].filesToCheck...)
}
```
And for specific resource, the whole switch case could be replaced with
```
resources = append(resources, resourcesMap[resource].k8sResourcesToCheck...)
filePaths = append(resources, resourcesMap[resource].filesToCheck...)
```

(with some handling to avoid nil slice being appended)
That way, adding a resource and/or a file to check would just need completing this map

Reply:
Hm... I didn't think someone would run two resources; if they did, there is always an option to specify the correct resource explicitly. Do you think it would be the case when the user has two resources, and we still need data from both (or more) of them?

Reply:
I thought about a scenario where some customers had a "project" namespace, like "reporting", with maybe multiple technologies under same namespace.

If there's a pxc + a pg cluster, I think one could be missed with "resource=auto" currently. If we miss the one we wanted, it's a bit disappointing

The map solution could enable to catch everything, and would reduce the amount of if/else/switch code.
It would replace it with something like
```
resourcesMap := map[string]resourceData{
    "pxc": resourceData{ 
        filesToCheck: []string{"var/lib/mysql/mysqld-error.log", <and so on>}, 
        k8sResourcesToCheck : []string{"perconaxtradbclusterbackups", "perconaxtradbclusterrestores", "perconaxtradbclusters"},
    },
    "pg": ....
```

I'm thinking of a map here because it does not look so easy to add a loop with the current code to add multiple resources at once
I think it can avoid potential back and forth to investigate, and would enable to avoid touching code when modifying/adding resources later on

Reply:
Thank you, @ylacancellera ! Let's make it another Jira and PR.